### PR TITLE
add sql credential pass through doc

### LINKED ISSUE #882: [DOC] Add instruction of SQL Credential Pass through with Access Token
### Willingness to contribute

Yes. I can contribute for this issue independently.

### URL(s) with the issue

_No response_

### Description of proposal (what needs changing)

token auth mode can be used to realize sql credential pass through scenarios.

### PR DESCRIPTION
**Pull request by Yuqing-cat**
Signed-off-by: Yuqing Wei <weiyuqing021@outlook.com>

## Description

Resolves #882

## Does this PR introduce any user-facing changes?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: docs/how-to-guides/feathr-credential-passthru.md [hunk 1]
```diff
@@ -34,4 +34,13 @@ client.materialize_features(settings, allow_materialize_non_agg_feature=True, ex
...
 In this code block, replace the `appId`, `clientSecret`, and `tenant` placeholder values in this code block with the values that you collected while completing the first step.
 
-3. Don't forget your other configuration settings, such as the ones that are specific to Feathr in [Feathr Job Configuration during Run Time](./feathr-job-configuration.md).
\ No newline at end of file
+3. Don't forget your other configuration settings, such as the ones that are specific to Feathr in [Feathr Job Configuration during Run Time](./feathr-job-configuration.md).
+
+4. Azure SQL Database Credential pass through is also supported. To achieve so you need to pass your token to environment variables and set `auth` parameter to `TOKEN` in `JdbcSource` or `JdbcSink`. For example:
+```python
+name = 'output'
+sink = client.JdbcSink(name, some_jdbc_url, dbtable, "TOKEN")
```
Comment (blrchen) [inline docs/how-to-guides/feathr-credential-passthru.md:42]: Suggest change to sink = client.JdbcSink(name, some_jdbc_url, dbtable, auth="TOKEN") to improve code readability

File: docs/how-to-guides/jdbc-cosmos-notes.md [hunk 2]
```diff
@@ -62,6 +62,20 @@ client.get_offline_features(...)
 
 These values will be automatically passed to the Feathr core when submitting the job.
 
+If you want to use token, the code will be like this:
+```
+src1_name="source1"
+source1 = JdbcSource(name=src1_name, url="jdbc:...", dbtable="table1", auth="TOKEN")
+anchor1 = FeatureAnchor(name="anchor_name",
```
Comment (blrchen) [inline docs/how-to-guides/jdbc-cosmos-notes.md:69]: Could the 1 postfix in src1, source1, anchor1 get removed?

Comment (Yuqing-cat) [reply docs/how-to-guides/jdbc-cosmos-notes.md:69]: This part aligns with the `USERPASS` sample. I think the purpose of 1 postfix is to indicate that there could be multiple sources. let me update it with 2 then.

File: docs/how-to-guides/feathr-credential-passthru.md (hunk 1 above)
Comment (Yuqing-cat) [reply docs/how-to-guides/feathr-credential-passthru.md:42]: sure, will update.

File: docs/how-to-guides/jdbc-cosmos-notes.md [hunk 3]
```diff
@@ -62,6 +62,20 @@ client.get_offline_features(...)
 
 These values will be automatically passed to the Feathr core when submitting the job.
 
+If you want to use token, the code will be like this:
+```python
+src_name="source_name"
```
Comment (xiaoyongzhu) [inline docs/how-to-guides/jdbc-cosmos-notes.md:67]: Where should user get this `auth="TOKEN"` part?

Comment (Yuqing-cat) [reply docs/how-to-guides/jdbc-cosmos-notes.md:67]: will add a snippet of `how to generate token` and add a skipped test case for now.

**Review by blrchen — APPROVED** [review]

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**